src/NodaTime/Calendars/WeekYearRule.cs

```diff
@@ -0,0 +1,199 @@
+        /// week 52 of week-year 2010. Conversely, December 31st 2012 is a Monday,
+        /// so is part of week 1 of week-year 2013.
+        /// </para>
+        /// <value>A <see cref="WeekYearRule"/> consistent with ISO-8601.</value>
+        public static WeekYearRule Iso { get; } = isoBasedRules[3];
+
+        // TODO: public static WeekYearRule FromBclRule(CalendarWeekRule rule, IsoDayOfWeek firstDayOfWeek) => null;
+        // TODO: public static WeekYearRule FromBclRule(CalendarWeekRule rule, DayOfWeek firstDayOfWeek)
+        //    => FromBclRule(rule, BclConversions.ToIsoDayOfWeek(firstDayOfWeek));
+
+        // TODO: Make IsoBasedWeekYearRule more flexible, allowing non-Monday-to-Sunday weeks?
+
+        /// <summary>
+        /// Creates a <see cref="LocalDate" /> from a given week-year, week within that week-year,
+        /// and day-of-week, for the specified calendar system.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Valid values for <paramref name="weekYear"/> and <paramref name="weekOfWeekYear"/> will
+        /// depend on the rule and calendar system. Rules will usually permit any week-year which is also a valid calendar year,
+        /// and sometimes the value one less than the minimum calendar year and one more than the maximum
+        /// calendar year. The valid weeks within that week-year may vary as some years are longer than others
+        /// in terms of weeks - and the first valid week-year and last valid week-year usually only contain a few
+        /// weeks each. In most applications this is unlikely to be an issue as the range of valid dates is far
+        /// wider than is usually required.
+        /// </para>
+        /// <para>
+        /// Wherever reasonable, implementations should ensure that all valid dates
+        /// can be constructed via this method. In other words, given a <see cref="LocalDate"/> <code>date</code>,
+        /// <code>rule.CreateLocalDate(rule.GetWeekYear(date), rule.GetWeekOfWeekYear(date), date.IsoDayOfWeek, date.Calendar)</code>
+        /// should always return <code>date</code>. This is true for all rules within Noda Time, but third party
+        /// implementations may choose to simplify their implementations by restricting them to appropriate portions
+        /// of time.
+        /// </para>
+        /// </remarks>
+        /// <param name="weekYear">The week-year of the new date.</param>
+        /// <param name="weekOfWeekYear">The week of week-year of the new date.</param>
+        /// <param name="dayOfWeek">The day-of-week of the new date.</param>
+        /// <param name="calendar">The calendar system for the date.</param>
+        /// <exception cref="ArgumentOutOfRangeException">The parameters do not combine to form a valid date.</exception>
+        /// <returns>A <see cref="LocalDate"/> corresponding to the specified values.</returns>
+        public abstract LocalDate CreateLocalDate(
```

Comment:
Seems a little odd to call this "CreateXxx" when LocalDate is a value type. `GetLocalDate`, perhaps?